cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/BusinessMetadataDataset.java

```diff
@@ -461,13 +465,27 @@ private void addNamespaceIdToKey(MDSKey.Builder builder, Id.NamespacedId namespa
     throw new IllegalArgumentException("Illegal Type " + type + " of metadata source.");
   }
 
+  private String getNamespaceId(MDSKey key) {
```

Comment:
oh oops no I may have got that wrong. Can you instead change the comment two lines below to split the ``[targetId]`` into its components? Or another comment mentioning namespaceId is the first part of ``[targetId]``